fileServer.js

```diff
@@ -0,0 +1,113 @@
+ 'use strict';
+
+ var http = require('http');
+ var fs = require('fs');
+ var path = require('path');
+ var url = require('url');
+ var rootDir = __dirname + '/public';
+
+ const mimeType = {
+   'html': 'text/html',
+   'jpeg': 'image/jpeg',
+   'jpg': 'image/jpeg',
+   'png': 'image/png',
+   'js': 'text/javascript',
+   'css': 'text/css'
+ };
+
+ function send404Request(res) {
+   fs.readFile('public/404.html', 'UTF-8', function(err, html) {
+<<<<<<< HEAD
+     if (err) {
+       res.writeHead(500);
+       res.write("Internal Server Error");
+     } else {
+       res.writeHead(404, {
+         'Content-type': mimeType
+       });
+       res.end(html);
+     }
+=======
+     res.setHeader('Content-type', mimeType);
+     res.end(html);
+>>>>>>> 945100f8f04e33242f280c0f03422c5c690fe44b
+   });
+ }
+
+ function onRequest(req, res) {
+<<<<<<< HEAD
+   console.log('request starting....');
+=======
+   console.log('request starting...');
+>>>>>>> 945100f8f04e33242f280c0f03422c5c690fe44b
+
+   var filename;
+   var pathname = url.parse(req.url).pathname;
+   filename = path.join(rootDir, pathname);
+   console.log('Filename: ' + filename);
+<<<<<<< HEAD
+   var stream = fs.createReadStream(filename);
+=======
+>>>>>>> 945100f8f04e33242f280c0f03422c5c690fe44b
+
+   if (req.method !== 'GET') {
+     res.writeHead(405, {
+       "Content-type": "text/plain"
+     });
+     res.write(`HTTP method ${req.method} not yet supported`);
+     res.end();
+     return
+   }
+
+   if (req.url === '/') {
+<<<<<<< HEAD
+     fs.readFile('public/index.html', 'UTF-8', function(err, html) {
+       if (err) {
+         res.writeHead(500);
+         res.write("Internal Server Error");
```

Comment:
Tell the user why the error occurred by returning error text in the body and tell the sysadmin why it occurred by logging it with console.log - why means logging the error message itself (err variable)